test/support/page_objects/header_page.js

```diff
@@ -144,6 +144,24 @@ export default class HeaderPage {
     .click();
   }
 
+  async waitUntilLoadingHasFinished() {
+    try {
+      await this.isGlobalLoadingIndicatorVisible();
+    } catch (exception) {
+      if (exception.name === 'NoSuchElement') {
+        // selenium might just have been too slow to catch it
+      } else {
+        throw exception;
+      }
+    }
+    await this.isGlobalLoadingIndicatorHidden();
+  }
+
+  isGlobalLoadingIndicatorVisible() {
+    return this.remote.setFindTimeout(defaultFindTimeout)
+      .findByCssSelector('[data-test-subj="globalLoadingIndicator"]:not(.ng-hide)');
```

Comment:
This can just use `findTestSubject()`. The reason it couldn't be used below is because it uses `findDisplayedByCssSelector()` under the hood.